include/swift/AST/Evaluator.h

```diff
@@ -384,8 +373,20 @@ class Evaluator {
     FrontendStatsTracer statsTracer = make_tracer(stats, request);
     if (stats) reportEvaluatedRequest(*stats, request);
 
+    recorder.beginRequest(activeReq);
+
     auto &&r = getRequestFunction<Request>()(request, *this);
-    reportEvaluatedResult<Request>(request, r);
+
+    recorder.endRequest(activeReq);
+
+    handleDependencySourceRequest<Request>(request);
+    handleDependencySinkRequest<Request>(request, r);
```

Comment:
I believe we can fold all of this into `DependencyRecord::endRequest` - and because there are no combined sources and sinks it should be possible to branch on the caching strategy bits in the templates.

Reply:
Ok. Can I address this in a follow-up?